lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb

```diff
@@ -30,7 +30,7 @@ def install!
               # cause an App Store rejection because frameworks cannot be
               # embedded in embedded targets.
               #
-              create_embed_frameworks_script if target.includes_frameworks? && !target.requires_host_target?
+              create_embed_frameworks_script if !target.requires_host_target?
```

Comment:
I don't think this is the way to go - `target.includes_frameworks?` is the place that should be deciding if there are still any frameworks which require the embed frameworks script.

The logic for that is here:

https://github.com/CocoaPods/CocoaPods/blob/af83e4713771bc98dd7f903cc81c4fbc8866812c/lib/cocoapods/target/aggregate_target.rb#L227-L232

I recommend looking at that section of code to see if we are incorrectly determining that an aggregate target still has framework dependencies

Reply:
There is nothing wrong with that method. It's not about incorrectly detecting the number of frameworks. Simply removing the last framework, the modified line of code makes it skip the script file update, so the framework related line of code doesn't get removed.

Reply:
For example, you install `TwitterKit`, it adds:

```rb
if [[ "$CONFIGURATION" == "Debug" ]]; then
  install_framework "${PODS_ROOT}/TwitterCore/iOS/TwitterCore.framework"
  install_framework "${PODS_ROOT}/TwitterKit/iOS/TwitterKit.framework"
fi
if [[ "$CONFIGURATION" == "Release" ]]; then
  install_framework "${PODS_ROOT}/TwitterCore/iOS/TwitterCore.framework"
  install_framework "${PODS_ROOT}/TwitterKit/iOS/TwitterKit.framework"
fi
```

you then remove `TwitterKit`, I think it removes it from the project(s), and then the `includes_frameworks?` method returns `false` and the `TwitterCore.framework` and `TwitterKit.framework` references incorrectly get left in the script.

Reply:
@amorde, Was I clear enough? 😃